python-package/xgboost/config.py

```diff
@@ -132,7 +140,7 @@ def get_config():
     set_config: Set global XGBoost configuration
     get_config: Get current values of the global configuration
             """)
-def config_context(**new_config):
+def config_context(**new_config: Any) -> Iterator[None]:
```

Comment:
Is it a more general generator or an iterator?

Reply:
At first glance, it seems that `Generator` is more proper than `Iterator`, because `config_context` is implemented as a `Generator` here. However, as `config_context` is essentially a context manager, it's not wrong to annotate the return type as an `Iterator`. In fact, `Iterator` is more general than `Generator`, as `Generator` is a subclass of `Iterator`:
`assert issubclass(Generator, Iterator)`

For mypy, however, the choice remains controversial as tracked by this open issue: https://github.com/python/typeshed/issues/2772

On a side note, mypy says
`The return type of a generator function should be "Generator" or one of its supertypes`
when I use a wrong type hint for the function.

Reply:
> In fact, Iterator is more general than Generator,

Thank you for sharing. I need to remind myself about coroutines. In that case Iterator should be fine.